File: ClientApp/src/app/features/reservations/classes/resources/passenger-write-resource.ts
import { Guid } from "guid-typescript"

export class PassengerWriteResource {

    id: number
    reservationId: Guid
    occupantId: number
    nationalityId: number
    genderId: number
    lastname: string
    firstname: string
    birthDate: string
    specialCare: string
    remarks: string
    isCheckedIn: boolean

}

File: API/Features/Reservations/Resources/Passengers/PassengerWriteResource.cs
using System;

namespace API.Features.Reservations {

    public class PassengerWriteResource {

        public int Id { get; set; }
        public Guid? ReservationId { get; set; }
        public int GenderId { get; set; }
        public int NationalityId { get; set; }
        public int OccupantId { get; set; }
        public string Lastname { get; set; }
        public string Firstname { get; set; }
        public string Birthdate { get; set; }
        public string SpecialCare { get; set; }
        public string Remarks { get; set; }
        public bool IsCheckedIn { get; set; }

    }

}